docs/go-helloworld.md

```diff
@@ -0,0 +1,170 @@
+
+```
+go mod init github.com/{your-username}/temporal-go-tutorials/helloworld
+```
+
+`github.com/{your-username}/temporal-go-tutorials/helloworld` is the name of your Go module which you will use later on.
+
+Add the Temporal Go SDK as a dependency:
+
+```
+go get -u go.temporal.io/sdk
+```
+
+You should now have the following files in your project:
+
+- go.mod
+- go.sum
+
+## Build "Hello World!" app
+
+Now we are ready to create our Temporal Workflow app. The following code snippets are from the [temporalio/go-samples](https://github.com/temporalio/go-samples/tree/master/helloworld) repository and can be copied and pasted directly into your Java files.
+
+### Activity
+
+First, let's define our Activity. In your project directory create the file "activity.go":
+
+```
+touch activity.go
+```
+
+Copy the following code and paste it into activity.go:
+
+<!--SNIPSTART go-helloworld-sample-activity-->
+<!--SNIPEND-->
+
+Activities are meant to handle non-deterministic code that could result in unexpected results or errors. For this tutorial all we are doing is taking a string, appending it to "Hello", and returning it back to the Workflow.
+
+### Workflow
+
+In your project directory create the file "workflow.go":
+
+```
+touch workflow.go
+```
+
+Copy the following code and paste it into workflow.go:
+
+<!--SNIPSTART go-helloworld-sample-workflow-->
+<!--SNIPEND-->
+
+As you can see in the code above, a Workflow is simply a function that organizes the sequence of Activity executions. Opposed to an Activity, a Workflow is deterministic in nature. It is where you can specify Activity options, such as what to do if the Activity results in an error or when to time out the Activity execution. Since our Activity is just formatting strings we will only specify the required timeout. We have also explicitly defined the name of the Task Queue to which the Workflow and Activity Tasks will be sent. This is important because the Worker will register itself to listen to the same Task Queue.
+
+### Starter
+
+There are two ways to start a Workflow, either via the CLI or via a starter program. In this tutorial we will start our Workflow by invoking it with a starter program.
+
+In your project directory create a new directory called "starter" and add a new file to it called "main.go":
+
+```
+mkdir starter
+touch starter/main.go
+```
+
+Copy the following code and paste it into starter/main.go, making sure to replace the import path `"github.com/temporalio/temporal-go-samples/helloworld"` with your Go module import path:
+
+<!--SNIPSTART go-helloworld-sample-workflow-starter-->
+<!--SNIPEND-->
+
+In the code above, we are creating a Temporal client and using it to host the Workflow with the Temporal server. Said another way, this starter program is sending the Workflow to the Temporal server where it will be broken into Tasks. The Tasks will be sent to the Task Queue that we supplied in the Workflow options.
```

Comment:
I think "host workflow" is not really a good term in this case.

Reply:
> Said another way, this starter program is sending the Workflow to the Temporal server where it will be broken into Tasks. The Tasks will be sent to the Task Queue that we supplied in the Workflow options.

This sentence contains too many new terms. Workflow is not broken into tasks by the server. We also didn't define what tasks are as we only had workflows and activities. We either need to define the relationship or do not mention tasks.